WireAPIService/Sources/WireAPIService/Endpoints/BackendInfo/APIService+BackendInfo.swift

```diff
@@ -0,0 +1,35 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+extension APIService {
+    public func getBackendInfo(for version: APIVersion) async throws -> BackendInfoModel {
+        let endpoint = Endpoints.BackendInfo()
```

Comment:
I really think while we refactor the API layer we should also move away from switching over the API version and instead resolve once at launch and then create the suitable API classes at initialisation time.

Reply:
To avoid duplication of the endpoints and versions we could also just pass the api versions as configuration:
```swift
    struct BackendInfo: Endpoint {

        let version: APIVersion

        let path: String = "/api-version"

        let method: HTTPMethod = .get

        let queryItems: [URLQueryItem]? = nil

        var body: Data? {
            switch version {
            case .v0, .v1:
                return nil
            case .v2, .v3, .v4, .v5, .v6:
                return Data()
            }
        }
    }
```

Reply:
But if it's a completely different endpoint with different models, I would rather create a new class/struct.